From a800c2e9ce394b076eb4e15ae1cfa9a197e7aabb Mon Sep 17 00:00:00 2001
From: Jeremy Olexa <darkside@gentoo.org>
Date: Wed, 7 Dec 2011 15:58:25 +0000
Subject: [PATCH] s:/etc/make.profile:/etc/portage/make.profile:g - Use new location

Take 2, specify target symlink better and create /etc/portage first
(fails with stage2 otherwise)
---
 modules/generic_stage_target.py     |    7 ++++---
 targets/stage1/stage1-chroot.sh     |    2 +-
 targets/stage1/stage1-controller.sh |    2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/modules/generic_stage_target.py b/modules/generic_stage_target.py
index 55baea7..6e0bc33 100644
--- a/modules/generic_stage_target.py
+++ b/modules/generic_stage_target.py
@@ -813,11 +813,12 @@ class generic_stage_target(generic_target):
 			# TODO: zmedico and I discussed making this a directory and pushing
 			# in a parent file, as well as other user-specified configuration.
 			print "Configuring profile link..."
-			cmd("rm -f "+self.settings["chroot_path"]+"/etc/make.profile",\
+			cmd("rm -f "+self.settings["chroot_path"]+"/etc/portage/make.profile",\
 					"Error zapping profile link",env=self.env)
-			cmd("ln -sf ../usr/portage/profiles/"+\
+			cmd("mkdir "+self.settings["chroot_path"]+"/etc/portage/")
+			cmd("ln -sf ../../usr/portage/profiles/"+\
 				self.settings["target_profile"]+" "+\
-				self.settings["chroot_path"]+"/etc/make.profile",\
+				self.settings["chroot_path"]+"/etc/portage/make.profile",\
 				"Error creating profile link",env=self.env)
 			touch(self.settings["autoresume_path"]+"config_profile_link")
 
diff --git a/targets/stage1/stage1-chroot.sh b/targets/stage1/stage1-chroot.sh
index 79f127f..976515d 100644
--- a/targets/stage1/stage1-chroot.sh
+++ b/targets/stage1/stage1-chroot.sh
@@ -14,7 +14,7 @@ if [ -z "${clst_buildpkgs}" ]
 then
 	echo "Your profile seems to be broken."
 	echo "Could not build a list of build packages."
-	echo "Double check your /etc/make.profile link and the 'packages' files."
+	echo "Double check your /etc/portage/make.profile link and the 'packages' files."
 	exit 1
 fi
 
diff --git a/targets/stage1/stage1-controller.sh b/targets/stage1/stage1-controller.sh
index 5625877..1f72067 100644
--- a/targets/stage1/stage1-controller.sh
+++ b/targets/stage1/stage1-controller.sh
@@ -13,7 +13,7 @@ case $1 in
 		
 		# Setup make.conf and make.profile link in "ROOT in chroot":
 		copy_to_chroot ${clst_chroot_path}/etc/make.conf /${clst_root_path}/etc
-		copy_to_chroot ${clst_chroot_path}/etc/make.profile \
+		copy_to_chroot ${clst_chroot_path}/etc/portage/make.profile \
 			/${clst_root_path}/etc
 
 		# Enter chroot, execute our build script
-- 
1.7.3.4

